Source/DafnyCore/Rewriters/AllOpaqueRevealStmtInserter.cs

```diff
@@ -0,0 +1,531 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Linq.Expressions;
+using Microsoft.Boogie;
+using Microsoft.Dafny.Plugins;
+
+namespace Microsoft.Dafny; 
+
+public class AllOpaqueRevealStmtInserter : IRewriter {
+  public AllOpaqueRevealStmtInserter(ErrorReporter reporter) : base(reporter) { }
+
+  internal override void PostResolveIntermediate(ModuleDefinition moduleDefinition) {
+    Contract.Requires(moduleDefinition != null);
+
+    foreach (var decl in moduleDefinition.TopLevelDecls) {
+      if (decl is TopLevelDeclWithMembers cl) {
+        foreach (var member in cl.Members) {
+          if (member is ICallable and not ConstantField) {
+            var mem = (ICallable)member;
+            if (member is Function { ByMethodDecl: not null } f) {
+              mem = f.ByMethodDecl;
+            }
+
+            if (mem is Method { Body: not null } method) {
+              AddMethodReveals(method, Reporter);
+            } else if (mem is Function { Body: not null } func) {
+              AddFunctionReveals(func, Reporter);
+            }
+          } else if (member is ConstantField { Rhs: not null } cf) {
+            var subExpressions = cf.Rhs.SubExpressions;
+
+            foreach (var expression in subExpressions) {
+              if (expression is FunctionCallExpr funcExpr) {
+                var func = funcExpr.Function;
+
+                if (isRevealedAlongPath(func, moduleDefinition, new List<ModuleDefinition> { func.EnclosingClass.EnclosingModuleDefinition })) {
+                  if (!func.IsOpaque && func.DoesAllOpaqueMakeOpaque(Options)) {
+                    var expr = cf.Rhs;
+
+                    var revealStmt0 = BuildRevealStmt(new FunctionReveal(func, new List<NameSegment>()),
```

Comment:
Can you please create an overload of BuildRevealStmt so that you don't need to create an object just to call it? The first argument could just be "func".